src/util.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Based on environment, get a reference to the Web Crypto API's SubtleCrypto interface.
+ * @returns An implementation of the Web Crypto API's SubtleCrypto interface.
+ */
+function getSubtleCrypto(): SubtleCrypto {
+	if (typeof window !== 'undefined' && window.crypto) {
+		return window.crypto.subtle;
+	}
+	if (typeof globalThis !== 'undefined' && globalThis.crypto) {
+		return globalThis.crypto.subtle;
+	}
+	if (typeof crypto !== 'undefined') {
+		return crypto.subtle;
+	}
+	if (typeof require === 'function') {
+		// Cloudflare Workers are doing what appears to be a regex check to look and
+		// warn for this pattern. We should never get here in a Cloudflare Worker, so
+		// I am being coy to avoid detection and a warning.
+		const cryptoPackage = 'node:crypto';
+		const crypto = require(cryptoPackage);
+		return crypto.webcrypto.subtle;
+	}
+	throw new Error('No Web Crypto API implementation found');
+}
+
+export const subtleCrypto = getSubtleCrypto();
+
+/**
+ * Converts different types to Uint8Array.
+ *
+ * @param value - Value to convert. Strings are parsed as hex.
+ * @param format - Format of value. Valid options: 'hex'. Defaults to utf-8.
+ * @returns Value in Uint8Array form.
+ */
+export function valueToUint8Array(
+	value: Uint8Array | ArrayBuffer | Buffer | string,
+	format?: string,
+): Uint8Array {
+	if (value == null) {
+		return new Uint8Array();
+	}
+	if (typeof value === 'string') {
+		if (format === 'hex') {
+			const matches = value.match(/.{1,2}/g);
+			if (matches == null) {
+				throw new Error('Value is not a valid hex string');
+			}
+			const hexVal = matches.map((byte: string) => Number.parseInt(byte, 16));
```

Comment:
This is kind of insane. Is this really the best approach 😆

Reply:
dude ... if you can find a better approach that works I'm all ears.  This is a copy/paste from the original code, just shifted over.  I tried doing some other clever things with `btoa` and `TextEncoder`, but the whole "hex string" thing threw me for a loop.